File: game-engine/src/main/java/com/animousen4/game/engine/core/repositories/UserRepository.java
package com.animousen4.game.engine.core.repositories;

import com.animousen4.game.engine.core.repositories.entities.UserEntity;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends CrudRepository<UserEntity, Long> {
    UserEntity findUserEntityById(long id);
}

File: auth-server-webflux/src/main/kotlin/com/animousen4/authserverwebflux/core/repo/UserRepository.kt
package com.animousen4.authserverwebflux.core.repo

import com.animousen4.authserverwebflux.core.entities.UserEntity
import org.springframework.data.r2dbc.repository.R2dbcRepository
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono
@Repository
interface UserRepository : R2dbcRepository<UserEntity, Long> {
    fun getUserEntityByEmail(email: String) : Mono<UserEntity>

}